artemis-ra/src/main/java/org/apache/activemq/artemis/ra/ActiveMQRAManagedConnection.java

```diff
@@ -336,7 +336,7 @@ public void associateConnection(final Object obj) throws ResourceException {
 
    public void checkTransactionActive() throws JMSException {
       // don't bother looking at the transaction if there's an active XID
-      if (!inManagedTx && tsr != null) {
+      if (!inManagedTx && tsr != null && tsr.getTransactionKey() != null) {
```

Comment:
This change doesn't appear related to the `transactionTimeout` change. Why is this necessary?

Reply:
a leftover